tests/test_trainer.py

```diff
@@ -410,6 +410,74 @@ class CurrentTestModel(
         tutils.run_prediction(dataloader, trainer.model)
 
 
+def test_resume_from_checkpoint_epoch_restored(tmpdir):
+    """Verify resuming from checkpoint runs the right number of epochs"""
+    import types
+
+    tutils.reset_seed()
+
+    hparams = tutils.get_hparams()
+
+    def new_model():
+        # Create a model that tracks epochs and batches seen
+        model = LightningTestModel(hparams)
+        model.num_epochs_seen = 0
+        model.num_batches_seen = 0
+
+        def increment_epoch(self):
+            self.num_epochs_seen += 1
+
+        def increment_batch(self, _):
```

Comment:
why do you have the _ there, maybe check also #776

Reply:
Since `on_batch_start` has the signature `(self, batch)` and I don't need `batch` its common to just call it `_`.
This isn't in a callback so I'm not sure how this relates to #776. I don't know how we could implement this using a callback.